services/securityService/authenticateRPCIdArg.js

```
@@ -0,0 +1,37 @@
+'use strict';
+
+const _ = require('lodash');
+const serviceUtils = require('../serviceUtils');
+
+/**
+ * Get the system parameters (represented by file 8989.3)
+ * @param  {Object}   incomingPatientId  Incoming Patient Id supplied by Native RPC Arguments
+ * @param  {Object}   context  Session context information associated with the service call.
+ * @param  {Function} callback Function called on process completion, with standard (err, result) signature.
+ * @return {Object}            Kernel System Parameters associated with the station ID in the context, or {} if not found
+ *
+ * @memberof securityService
+ * @instance
+ */
+
+function checkIfNativeArgMatchesSession(incomingPatientId, context) {
+    const nativeArg = parseInt(incomingPatientId, 10);
+    let patientId = _.get(context, 'patientId', false);
+
+    if (!patientId) {
+        return { error: 'The context object is missing a patient Id' };
```

Here, instead of returning a JSON object with an `error` field, let's return an Error object instead.

This would map better to what we eventually want to do, which is have an enumerated Error service which returns Errors based on name/code selection.

---

Got it!